library/src/main/kotlin/build/buf/connect/impl/ProtocolClient.kt

```diff
@@ -52,16 +53,22 @@ class ProtocolClient(
         val serializationStrategy = config.serializationStrategy
         val requestCodec = serializationStrategy.codec(methodSpec.requestClass)
         try {
+            val requestMessage = if (config.getConfiguration != GETConfiguration.GETDisabled) {
+                // Use deterministic serialization when GET request configuration is set.
+                requestCodec.deterministicSerialize(request)
+            } else {
+                requestCodec.serialize(request)
```

Comment:
Is there a particular reason why we wouldn't just always deterministically serialize?

Reply:
Performance concerns primarily